File: cc-check/test/fixtures/rust/src/contracts.rs
//! @cc [author:spolu,label:product] rust-module-fixture
//! Documents a Rust module contract that is checked without declaration attachment.

/**
 * @cc [author:spolu,label:product] rust-account-fixture
 * Represents a Rust account used by the CLI integration tests.
 */
pub struct Account {
    /// @cc [author:spolu,label:product] rust-balance-fixture
    /// Provides a named field used to exercise nested Rust contract listing.
    balance: i64,
}

/// @cc [author:spolu,label:product] rust-account-impl-fixture
/// Defines behavior shared by methods in this account implementation.
impl Account {
    /// @cc [author:spolu,label:product] rust-payment-fixture
    /// Provides a method used to exercise implemented-type contract listing.
    #[must_use]
    pub fn pay(&mut self, amount: i64) -> i64 {
        self.balance -= amount;
        self.balance
    }
}

/// @cc [author:spolu,label:product] rust-state-fixture
/// Represents a separate Rust declaration for whole-file listing.
pub enum State {
    #[allow(dead_code)]
    /// @cc [author:spolu,label:product] rust-paid-variant-fixture
    /// Defines a contract on an attributed enum variant.
    Paid,
}

/// @cc [author:spolu,label:product] rust-payer-fixture
/// Defines a trait whose members can carry contracts.
pub trait Payer {
    /// @cc [author:spolu,label:product] rust-payer-signature-fixture
    /// Defines a contract on a trait method signature.
    fn pay(&mut self, amount: i64) -> i64;
}

/// @cc [author:spolu,label:product] rust-default-amount-fixture
/// Defines a constant item contract.
pub const DEFAULT_AMOUNT: i64 = 1;

File: cc-check/test/fixtures/php/contracts.php
<?php

#[Service]
/**
 * @cc [owner:spolu,label:product] php-invoice-service-fixture
 * Represents a PHP service used by the CLI integration tests.
 */
final readonly class InvoiceService
{
    /**
     * @cc [owner:spolu,label:product] php-amount-fixture
     * Provides a property used to exercise nested PHP contract listing.
     */
    public int $amount;

    /**
     * @cc [owner:spolu,label:product] php-default-amount-fixture
     * Defines a class constant in the PHP fixture.
     */
    public const DEFAULT_AMOUNT = 1;

    /**
     * @cc [owner:spolu,label:product] php-constructor-fixture
     * Defines a constructor with a promoted property.
     */
    public function __CONSTRUCT(
        /**
         * @cc [owner:spolu,label:product] php-promoted-property-fixture
         * Defines a contract on a constructor-promoted property.
         */
        private string $invoiceId,
    ) {}

    /**
     * @cc [owner:spolu,label:product] php-payment-fixture
     * Provides a keyword-named method used to exercise PHP parser recovery.
     */
    /**
     * @cc [owner:spolu,label:product] php-payment-result-fixture
     * Leaves the invoice in its expected paid state.
     */
    #[Transactional]
    public static function for(string $invoiceId): self
    {
        return new self();
    }
}

/**
 * @cc [owner:spolu,label:product] php-payer-fixture
 * Represents an interface used to exercise PHP declaration listing.
 */
interface Payer
{
    /**
     * @cc [owner:spolu,label:product] php-payer-signature-fixture
     * Defines a contract on an interface method.
     */
    public function pay(): void;
}

/**
 * @cc [owner:spolu,label:product] php-shared-behavior-fixture
 * Represents a trait used to exercise PHP declaration listing.
 */
trait SharedBehavior
{
    /**
     * @cc [owner:spolu,label:product] php-helper-fixture
     * Defines a contract on a trait method.
     */
    public function helper(): void {}
}

/**
 * @cc [owner:spolu,label:product] php-state-fixture
 * Represents an enum used to exercise PHP declaration listing.
 */
enum State: string
{
    /**
     * @cc [owner:spolu,label:product] php-paid-case-fixture
     * Defines a contract on an enum case.
     */
    case Paid = "paid";
}

/**
 * @cc [owner:spolu,label:product] php-report-fixture
 * Provides a separate PHP function for whole-file listing.
 */
function reportInvoices(): void {}

/* @cc this-is-not-phpdoc */
function ignoredOrdinaryComment(): void {}
